packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1225,10 +1236,24 @@ module.exports = (log, db, mailer, Password, config, customs, signinUtils, push,
             let devicesToNotify;
 
             return signinUtils.checkPassword(emailRecord, password, request.app.clientAddress)
-              .then((match) => {
+              .then(async (match) => {
                 if (! match) {
                   throw error.incorrectPassword(emailRecord.email, form.email);
                 }
+
+                if (config.subscriptions && config.subscriptions.enabled) {
+                  // Cancel all subscriptions before deletion, if any exist
+                  // Subscription records will be deleted from DB as part of account
+                  // deletion, but we have to trigger cancellation in payment systems
+                  const uid = emailRecord.uid;
+                  const subscriptions = await db.fetchAccountSubscriptions(uid) || [];
+                  for (const subscription of subscriptions) {
+                    const { subscriptionId } = subscription;
+                    await subhub.cancelSubscription(uid, subscriptionId);
+                    await db.deleteAccountSubscription(uid, subscriptionId);
```

Comment:
What's the worst thing that could happen if we crash after `cancelSubscription`, or after `deleteAccountSubscription`?  Do we risk leaving the account in a state where it cannot be successfully deleted if the user retries again later?